### PR TITLE
tools/depends: allow building GLES lib for x11 and wayland

### PR DESCRIPTION
**Pull request by lrusak**
This is a precursor to the multi-windowing stuff #18534 

This allows building the GLES library when using x11 or wayland

wayland version bump is needed because it now provides `wayland-egl-backend` needed by mesa

This also changes so that the libva library is built for all platforms.

### CODE REVIEW

**Comment by lrusak** [comment]
jenkins build this please

**Comment by wsnipex** [comment]
It doesn't make sense to build 3 different versions of nesa and libva that will even overwrite eaxh other. Why not build it once with support fir all platforms?

**Comment by lrusak** [comment]
> It doesn't make sense to build 3 different versions of mesa and libva that will even overwrite each other. Why not build it once with support for all platforms?

That is the direction I will be moving when we support multi windowing. One step at a time :smiley:

**Comment by wsnipex** [comment]
this does not depend on multi-windowing. Please explain why you can't build it once with all options enabled.

**Comment by lrusak** [comment]
> this does not depend on multi-windowing. Please explain why you can't build it once with all options enabled.

I didn't say it depends on multi-windowing. We can support building all the platforms together when multi-windowing comes. This is just a precursor to get GLES building for each platform 😊

**Comment by wsnipex** [comment]
I'm talking about dependencies, not kodi. It's wrong to build mesa and libva with different options, depending on the chosen windowing system.
Just always build those with support for all windowing platforms.

**Comment by a1rwulf** [comment]
> I'm talking about dependencies, not kodi. It's wrong to build mesa and libva with different options, depending on the chosen windowing system.
> Just always build those with support for all windowing platforms.

I don't really agree.
Why would you want to build mesa with X11 dependencies when you want to build for GBM?

**Comment by wsnipex** [comment]
the reason is simple: depends install to 1 target dir. 
If you first build kodi with x11, you'd end up with mesa and libva only supporting X11.
Now you want kodi wayland, but you'll have to rebuild depends instead of just kodi.
GBM? Rebuild depends once more . . .

Just look at how distros do it: They enable all options and the user/downstream lib uses what it needs.

**Review by wsnipex — CHANGES_REQUESTED** [review]
depends must not be built for specific windowing systems only

**Comment by a1rwulf** [comment]
Still, if you are on a system with wayland only, you cannot even compile the depends then, because you need to install X11 to do that - which you most likely don't want.
Anyway, just my 2c - I'm not using the depends build - so I'll let you argue about it with lrusak.

**Comment by lrusak** [comment]
> depends must not be built for specific windowing systems only

But we already do this, I haven't changed the behaviour here.

**Comment by lrusak** [comment]
So what's the path forward here? Do we not allow the flexibility of building for each platform individually? I plan on implementing the ability to build depends for all window systems together with the multi-windowing PR.

**Comment by wsnipex** [comment]
if you want to avoid adding X11 to depends you could also just get rid of the windowing specific disables for mesa and libva.
Just let them autodetect what's available.
And add a note to docs that you need X11 headers **on your build machine** if you want X11 support.

File: tools/depends/native/pugixml/Makefile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ARCHIVE=$(SOURCE).tar.gz
+BASE_URL=https://github.com/zeux/pugixml/releases/download/v$(VERSION)
+
+APP=$(PLATFORM)/pugixml
+
+# Only this package uses CMake for build on native at the moment,
+# so there is no separate toolchain file. Still we have to unset
+# the CMAKE_TOOLCHAIN_FILE, which is part of $(CMAKE) and set to
+# the target toolchain file.
+CMAKE_OPTIONS := -DCMAKE_TOOLCHAIN_FILE= -DCMAKE_C_COMPILER="$(CC_BINARY_FOR_BUILD)" -DCMAKE_CXX_COMPILER="$(CXX_BINARY_FOR_BUILD)" -DCMAKE_C_FLAGS="$(NATIVE_CFLAGS)" -DCMAKE_CXX_FLAGS="$(NATIVE_CXXFLAGS)" -DCMAKE_EXE_LINKER_FLAGS="$(NATIVE_LDFLAGS)" -DCMAKE_INSTALL_LIBDIR=lib $(CMAKE_OPTIONS)
```
Comment (garbear) [inline tools/depends/native/pugixml/Makefile:19]:
```suggestion
CMAKE_OPTIONS := -DCMAKE_TOOLCHAIN_FILE= \
  -DCMAKE_C_COMPILER="$(CC_BINARY_FOR_BUILD)" \
  -DCMAKE_CXX_COMPILER="$(CXX_BINARY_FOR_BUILD)" \
  -DCMAKE_C_FLAGS="$(NATIVE_CFLAGS)" \
  -DCMAKE_CXX_FLAGS="$(NATIVE_CXXFLAGS)" \
  -DCMAKE_EXE_LINKER_FLAGS="$(NATIVE_LDFLAGS)" \
  -DCMAKE_INSTALL_LIBDIR=lib $(CMAKE_OPTIONS)
```

File: tools/depends/native/Makefile [hunk 2]
```diff
@@ -49,7 +49,8 @@ tar: xz automake
 python3: $(EXPAT) libffi pkg-config zlib
 setuptools: python3
 wayland-scanner: expat
-waylandpp-scanner: cmake
+waylandpp-scanner: cmake pugixml
```
Comment (wsnipex) [inline tools/depends/native/Makefile:52]:
```suggestion
waylandpp-scanner: cmake expat pugixml
```

**Comment by wsnipex** [comment]
looks like wayland-scanner can't find expat, which installs it's pc file to x86_64-linux-gnu-native/lib/pkgconfig
Either we have to fix it's install libdir or set PKG_CONFIG_PATH for wayland-scanner

File: tools/depends/native/pugixml/Makefile [hunk 3]
```diff
@@ -8,15 +8,13 @@ LIBNAME=pugixml
 VERSION=1.10
 SOURCE=$(LIBNAME)-$(VERSION)
 ARCHIVE=$(SOURCE).tar.gz
-BASE_URL=https://github.com/zeux/pugixml/releases/download/v$(VERSION)
 
-APP=$(PLATFORM)/pugixml
+APP=$(PLATFORM)/build/libpugixml.a
```
Comment (garbear) [inline tools/depends/native/pugixml/Makefile:12]: To match the other makefiles this should be `LIBNAME` I think, or something with `LIB` instead of `APP`

**Comment by lrusak** [comment]
This finally builds again, however, I've had to disable x11 in mesa for arm builds (I'll have to cross compile the x11 libs or something, meh).

I've also included a few other various commits that fix some depends issues. If desired I can split them out into another PR.

**Review by wsnipex — APPROVED** [review]

**Pull request merged**